advanced/mysql_lambda_strategy/main.tf

```
@@ -0,0 +1,160 @@
+provider "sym" {
+  org = "sym-example"
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+data "aws_region" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+# Optionally set up a database to use for testing the integration
+module "db" {
+  source = "./mysql_db"
+  count  = var.db_enabled ? 1 : 0
+
+  tags = var.tags
+}
+
+############ Give Sym Runtime Permissions to execute your AWS Lambda ##############
+
+# Creates an AWS IAM Role that the Sym Runtime can use for execution
+# Allow the runtime to assume roles in the /sym/ path in your AWS Account
+module "runtime_connector" {
+  source  = "symopsio/runtime-connector/aws"
+  version = ">= 1.0.0"
+
+  environment = "main"
+
+  tags = var.tags
+}
+
+# An Integration that tells the Sym Runtime resource which AWS Role to assume
+# (The AWS Role created by the runtime_connector module)
+resource "sym_integration" "runtime_context" {
+  type = "permission_context"
+  name = "runtime-main"
+
+  external_id = module.runtime_connector.settings.account_id
+  settings    = module.runtime_connector.settings
+}
+
+# The AWS IAM Resources that enable Sym to invoke your Lambda functions.
+module "lambda_connector" {
+  source  = "symopsio/lambda-connector/aws"
+  version = ">= 1.0.0"
+
+  environment       = "main"
+  lambda_arns       = [module.mysql_lambda_function.lambda_function_arn]
+  runtime_role_arns = [module.runtime_connector.settings.role_arn]
+
+  tags = var.tags
+}
+
+# The Integration your Strategy uses to invoke Lambdas.
+# It points to to the AWS IAM resources created by the `lambda_connector` module.
+# This integration provides your Strategy the permissions needed to invoke your Lambda.
+resource "sym_integration" "lambda_context" {
+  type = "permission_context"
+  name = "lambda-context-main"
+
+  external_id = module.lambda_connector.settings.account_id
+  settings    = module.lambda_connector.settings
+}
+
+############ Lambda Strategy Setup ##############
+
+# A target AWS Lambda that will be invoked on escalate and de-escalate.
+# The `name` will be used in the lambda to decide which resource to manage access to
+resource "sym_target" "readonly" {
+  type  = "aws_lambda_function"
+  name  = "readonly"
+  label = "Read Only"
+
+  settings = {
+    # `type=aws_lambda_function` sym_targets have a required setting `arn`
+    # which must be the ARN of the AWS Lambda that will be invoked on escalate and de-escalate
+    arn = module.mysql_lambda_function.lambda_function_arn
+  }
+}
+
+# The Strategy your Flow uses to manage access
+resource "sym_strategy" "lambda" {
+  type = "aws_lambda"
+  name = "main-lambda-strategy"
```

Nit for consistency, since we've gotten feedback on it.

```suggestion
  name = "lambda-strategy-main"
```